cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MessagingMetricsProcessorService.java

```diff
@@ -221,75 +290,76 @@ private void processMetrics() {
         MessageFetcher fetcher = messagingService.prepareFetch(topicIdMetaKey.getTopicId());
         fetcher.setLimit(fetcherLimit);
         if (lastMessageId != null) {
+          if (LOG.isDebugEnabled()) {
+            LOG.debug("Start fetching from lastMessageId = {}", Bytes.toStringBinary(lastMessageId));
```

Comment:
You don't need the `lastMessageId` field anymore. Just get it from the `topicMessageIds` map.